### DISCUSSION TITLE
second fraction is not displayed

### DISCUSSION THREAD

**Discussion by KES777** [dbeaver/dbeaver#12841] [Q&A]
- PostgreSQL v13.1

#### Describe the problem you're observing:

data time is timestamptz

second fraction is not displayed
![image](https://user-images.githubusercontent.com/799021/119793535-594af080-bedf-11eb-814c-a36bd752c350.png)

#### Steps to reproduce, if exist:
I configure settings is described here: https://github.com/dbeaver/dbeaver/issues/12280#issuecomment-849442944

**Comment by Matvey16** [comment]
Hello. Try disabling date formatting like shown on the video. Also be sure that scale parameter is set according to your needs. It is under `scale` currently in UI, but will be under `precision` in 21.1.1.
You can change it by executing a query like that but with needed precision in the brackets `ALTER TABLE public.staff ALTER COLUMN column1 TYPE timestamptz(3) USING column1::timestamptz;`

![2021-06-03_17h09_44](https://user-images.githubusercontent.com/82543000/120660099-a73b9780-c48f-11eb-9a2f-c587572f3697.gif)
![2021-06-03_17h20_27](https://user-images.githubusercontent.com/82543000/120660504-10230f80-c490-11eb-980b-cc922bde3558.png)

**Comment by Matvey16** [comment]
Related to #12280

**Comment by KES777** [comment]
@Matvey16 As you can notice, I have already disabled that option which your show at your video. 
Did you see this screenshot? https://github.com/dbeaver/dbeaver/issues/12280#issuecomment-849442944

2. Also notice, this is screenshot from your video with your settings:
![image](https://user-images.githubusercontent.com/799021/121096427-ddb04400-c7fa-11eb-8aa4-7ae6638a0aae.png)

Fraction for seconds at first row is not displayed ;-)

Did you notice only `.8` instead of `.800000`?

**Comment by Matvey16** [comment]
@KES777 It looks like I got confused in two isuues.
Timestamp depends on many conditions. For example you can disable the date/time formatting (by checking it in properties like in my video) and then the look of the data will depend on various conditions. First is a scale parameter, it will cut the tail of your timestamp if it exceeds the scale. If the timestamp is inserted by a function, fraction will always be according to scale.
In both situations additional zeroes at the end will NOT be displayed because they are irrelivant.
That is why date/time formatting must be enabled to see those zeroes. If you enable the formating timpestamps will be shown differently. If your formatting looks like this `yyyy-MM-dd HH:mm:ss.ffffff` there will always be 6 digits after the dot. If your scale is 3 for example, 3 more zeroes will always be added at the end.

As you see. If you need additional zeroes at the end, you need to ENABLE formatting. So uncheck the option `Disable Daye/Time formatting`. The pattern for timestamp also changes the pattern for timestamtz.

Let me know, if you have any questions.

**Comment by KES777** [comment]
Finally I get it worked. IMHO: Interface is a bit weird. Also I do not understand what is precision and scale. Never saw that for `timestamp`

**Comment by Matvey16** [comment]
Precision is a parameter that decides how many numbers of milliseconds will be stored for this column. It is 6 by default if you don't specify it in paretesis. For example if you insert `12:00:02.23234345234` in `timestamp(3)` column, only `12:00:02.232` will be stored and even if you enable time formatting in DBeaver like this `HH:mm:ss.ffffff` it will show you `12:00:02.232000`
Scale parameter doesn't exist for timestamp in postgreSQL, but due to a bug that will be fixed in next version (and is already fixed in early acces version) the rpecision parameter is configured under scale column in the UI

![2021-06-03_17h20_27](https://user-images.githubusercontent.com/82543000/120660504-10230f80-c490-11eb-980b-cc922bde3558.png)
